### PR TITLE
request logger for chainweb-node

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/Utils/RequestLog.hs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    , _requestResponseLogDurationMicro = d
+    }
+
+-- | NOTE: this middleware should only be used for APIs that don't stream. Otherwise
+-- the logg may be delayed for indefinite time.
+--
+requestResponseLogger :: Logger l => l -> Middleware
+requestResponseLogger logger app req respond = do
+    let !reqLog = logRequest req
+    reqTime <- getTime Monotonic
```
Comment (gregorycollins) [inline src/Chainweb/Utils/RequestLog.hs]: Use the coarse monotonic clock: it's ~6x faster (as of the last time I benchmarked it), and you don't need nanosecond-level accuracy here

File: src/Chainweb/Utils/RequestLog.hs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    app req $ \res -> do
+        resTime <- getTime Monotonic
+        logFunctionJson logger Info
+            $ logRequestResponse reqLog res
+            $ (int $ toNanoSecs $ diffTimeSpec resTime reqTime) `div` 1000
+        respond res
```
Comment (gregorycollins) [inline src/Chainweb/Utils/RequestLog.hs]: So the time here is just processing time but doesn't cover the amount of time necessary to send the response body. Should you collect another time delta after `respond` finishes?

File: src/Chainweb/Utils/RequestLog.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/Utils/RequestLog.hs]: Accuracy is sufficient (~100 micro would be more than enough)?

I was just to lazy to check :-)

File: src/Chainweb/Utils/RequestLog.hs (hunk 2 above)
Comment (larskuhtz) [reply src/Chainweb/Utils/RequestLog.hs]: Oops, good catch. I'll fix it.

Usually, one wants both, "time to first byte" and "total time". One would also want to count the response bytes, but that requires some special casing on the different response body types. I don't think we have long-running streaming endpoints or very data heavy endpoints. So, let's keep it simple and add features as we need them.

File: src/Chainweb/Utils/RequestLog.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/Utils/RequestLog.hs]: It's seems that the `MonotonicCoarse` constructor is only available on Linux. Let's just use `Monotonic` for now.

**Pull request merged**